artemis-server/src/test/java/org/apache/activemq/artemis/core/server/impl/jdbc/JdbcLeaseLockTest.java

```diff
@@ -116,6 +122,22 @@ public void dropLockTable() throws Exception {
       jdbcSharedStateManager.close();
    }
 
+   @Test
+   public void shouldStillHeldLockOnTimeZoneChange() {
+      Assume.assumeThat(PropertySQLProvider.Factory.identifyDialect(dbConf.getJdbcDriverClassName()), CoreMatchers.is(PropertySQLProvider.Factory.SQLDialect.DERBY));
+      TimeZone.setDefault(TimeZone.getTimeZone(ZoneId.of("-1")));
+      final long expirationMillis = Duration.ofMinutes(30).toMillis();
+      final LeaseLock lock = lock(expirationMillis);
+      Assert.assertTrue("Must acquire the lock!", lock.tryAcquire());
+      TimeZone.setDefault(TimeZone.getTimeZone(ZoneId.of("+3")));
```

Comment:
this all look good to me, can't we merge this already?

Reply:
Yep this is not ready to merge yet

Reply:
@fvaleri I still have to decide how to handle testing this :)
But I've decided to use specific UTC get time query per supported DBs